app/controllers/posts_controller.rb

```diff
@@ -183,26 +183,25 @@ def short_link
   end
 
   def create
-    @manager_params = create_params
-    @manager_params[:first_post_checks] = !is_api?
-    @manager_params[:advance_draft] = !is_api?
-
-    manager = NewPostManager.new(current_user, @manager_params)
-
-    if is_api?
-      memoized_payload =
-        DistributedMemoizer.memoize(signature_for(@manager_params), 120) do
-          result = manager.perform
-          MultiJson.dump(serialize_data(result, NewPostResultSerializer, root: false))
-        end
+    manager_params = create_params
+    manager_params[:first_post_checks] = !is_api?
+    manager_params[:advance_draft] = !is_api?
+
+    manager = NewPostManager.new(current_user, manager_params)
+
+    json =
+      if is_api?
+        memoized_payload =
+          DistributedMemoizer.memoize(signature_for(manager_params), 120) do
+            MultiJson.dump(serialize_data(manager.perform, NewPostResultSerializer, root: false))
+          end
+
+        JSON.parse(memoized_payload)
+      else
+        serialize_data(manager.perform, NewPostResultSerializer, root: false)
```

Comment:
Ditto ^^ Inlined `result = manager.perform` since we don't need the temporary variable/state.